### PR TITLE
set up acceptance-tests and configure .travis.yml

### PR DESCRIPTION
**Pull request by jasson99**
set up acceptance test using nightwatch js and configure travis.yml to run the tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  - 8.12.0
+services:
+  - docker
+before_install:
+  - docker run -d -p 4444:4444 -p 5900:5900 -v /dev/shm:/dev/shm --name selenium selenium/standalone-chrome-debug
+before_script:
+  - cd parijat-frontend
+  - yarn
+  - yarn dev &
+  - sleep 30
```
Comment (skshetry) [inline .travis.yml]:
```suggestion
  - until curl -s localhost:8080; do sleep 5; done
```

Better to wait until the site is up. @jasson99 what do you think?

File: parijat-frontend/nightwatch.conf.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
```
Comment (skshetry) [inline parijat-frontend/nightwatch.conf.js]: Remove extra line.
```suggestion
```

File: parijat-frontend/test/acceptance/features/search.feature [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  Background:
+    Given the user has browsed to the homepage
+
+  Scenario: user should be able to search in english characters using translated text
+    When the user enters the english characters "munaa madan"
+    And the user selects the transliterated text
+    And the user searches for the literature
+    Then the search result should be displayed
+
+
```
Comment (skshetry) [inline parijat-frontend/test/acceptance/features/search.feature]:
```suggestion
```

File: parijat-frontend/test/acceptance/index.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const { setDefaultTimeout, After, Before } = require('cucumber')
+const { createSession, closeSession, startWebDriver, stopWebDriver } = require('nightwatch-api')
+
+setDefaultTimeout(60000)
+
+
```
Comment (skshetry) [inline parijat-frontend/test/acceptance/index.js]:
```suggestion
```

File: parijat-frontend/test/acceptance/features/transliteration.feature [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  So, I would be able to search using latin characters
+
+  Background:
+    Given the user has browsed to the homepage
+
+  Scenario: nepali string transliteration for the entered english character term should be displayed
+    When the user enters the english characters "munaa madan"
+    Then the transliteration should be "मुना मदन्"
+
+
```
Comment (skshetry) [inline parijat-frontend/test/acceptance/features/transliteration.feature]:
```suggestion
```

**Review by skshetry — COMMENTED** [review]
LGTM :+1:
Please try to find out if the repo has any opinion regarding with or without semicolon(`;`), and be consistent.

**Comment by jasson99** [comment]
cc please review @individual-it, @phil-davis, @HariBhandari07, @dpakach, @paurakhsharma

File: parijat-frontend/test/acceptance/stepDefinitions/searchContext.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+const userEntersTheEnglishOrNepaliCharacters = (searchTerm) => client.useXpath()
+	.waitForElementVisible(searchField)
+	.setValue(searchField, searchTerm)
+	.useCss();
+
+const userSearchesForTheLiterature = () => client.useXpath()
+	.click(searchButton)
+	.useCss();
+
+Given('the user has entered the english/nepali characters {string}', userEntersTheEnglishOrNepaliCharacters);
```
Comment (individual-it) [inline parijat-frontend/test/acceptance/stepDefinitions/searchContext.js]: english/nepali does not make any difference, so I would get rid of it

need to specify where does the user enter the characters

```suggestion
Given('the user has entered the string {string} into the main search field', userEntersTheEnglishOrNepaliCharacters);
```

File: parijat-frontend/test/acceptance/stepDefinitions/searchContext.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+When('the user enters the english/nepali characters {string}', userEntersTheEnglishOrNepaliCharacters);
```
Comment (individual-it) [inline parijat-frontend/test/acceptance/stepDefinitions/searchContext.js]: see comment above

**Review by individual-it — CHANGES_REQUESTED** [review]
only small wording issue

please use pageObjects (new commit)

File: parijat-frontend/package.json [hunk 8]
```diff
@@ -10,7 +10,8 @@
     "serve:gen-config": "preact serve --server config --dest superstatic.json",
     "dev": "preact watch",
     "test": "mocha --require babel-polyfill --require babel-register",
-    "lint": "eslint src"
+    "lint": "eslint src",
+    "test:e2e": "node_modules/cucumber/bin/cucumber-js --require test/acceptance/index.js --require test/acceptance/stepDefinitions"
```
Comment (individual-it) [inline parijat-frontend/package.json]: try using https://github.com/kozhevnikov/cucumber-pretty for a nicer output

**Review by individual-it — CHANGES_REQUESTED** [review]

File: parijat-frontend/test/acceptance/features/search.feature [hunk 9]
```diff
@@ -0,0 +1,29 @@
+Feature: search
+  As a user
+  I want to find the desired literature
+  So, I would be able to read them
+
+  Background:
+    Given the user has browsed to the homepage
+
+  Scenario: user should be able to search in english characters using translated text
```
Comment (phil-davis) [inline parijat-frontend/test/acceptance/features/search.feature]:
```suggestion
  Scenario: user should be able to search in english characters using transliterated text
```

File: parijat-frontend/test/acceptance/features/search.feature [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    When the user enters the string "munaa madan" into the main search field
+    And the user selects the transliterated text
+    And the user searches for the literature
+    Then the search result should be displayed
+
+  Scenario: no search result should be displayed when the user search for invalid literature
```
Comment (phil-davis) [inline parijat-frontend/test/acceptance/features/search.feature]:
```suggestion
  Scenario: no search result should be displayed when the user searches for invalid literature
```

File: parijat-frontend/test/acceptance/features/search.feature [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    And the user selects the transliterated text
+    And the user searches for the literature
+    Then the search result should not be displayed
+
+  Scenario: user should be able to search in nepali characters
+    When the user enters the string "मुना मदन्" into the main search field
+    And the user searches for the literature
+    Then the search result should be displayed
+
+  Scenario: no search result should be displayed when the user search for invalid literature
```
Comment (phil-davis) [inline parijat-frontend/test/acceptance/features/search.feature]:
```suggestion
  Scenario: no search result should be displayed when the user searches for invalid literature
```

File: parijat-frontend/test/acceptance/features/search.feature [hunk 12]
```diff
@@ -0,0 +1,29 @@
+Feature: search
+  As a user
+  I want to find the desired literature
+  So, I would be able to read them
```
Comment (phil-davis) [inline parijat-frontend/test/acceptance/features/search.feature]:
```suggestion
  So I can read them
```

File: parijat-frontend/test/acceptance/features/transliteration.feature [hunk 13]
```diff
@@ -0,0 +1,15 @@
+Feature: transliteration
+  As a user
+  I want to be able to transliterate latin characters into nepali characters
+  So, I would be able to search using latin characters
```
Comment (phil-davis) [inline parijat-frontend/test/acceptance/features/transliteration.feature]:
```suggestion
  So I can search using latin characters
```

**Comment by jasson99** [comment]
@individual-it, @phil-davis

File: parijat-frontend/package.json [hunk 14]
```diff
@@ -27,16 +28,20 @@
...
     "if-env": "^1.0.0",
     "mocha": "^5.2.0",
+    "nightwatch": "^1.2.4",
+    "nightwatch-api": "^2.3.0",
     "preact-cli": "^2.0.0",
     "superstatic": "^5.0.1"
   },
   "dependencies": {
     "babel-preset-env": "^1.7.0",
+    "cucumber-pretty": "^1.5.2",
```
Comment (phil-davis) [inline parijat-frontend/package.json]: Why is this added to the main dependencies section?
Can/should it be in `devDependencies`?

**Review by phil-davis — APPROVED** [review]

**Pull request merged**